### ISSUE TITLE
Stop button not needed if sendImmediately is set to true

### ISSUE THREAD

**Issue by sagrawal31** [Ziggeo/Android-Client-SDK#32]
Continuing discussion from https://github.com/Ziggeo/Android-Client-SDK/issues/30#issuecomment-597513590

Setting `.sendImmediately(false)` shows two buttons in the recorder:

1. "Stop & close button" which gives options to preview, re-record and submit the recording.
2. "Send & close" check button which directly submits the recording.

But conceptually, `sendImmediately` should not mix two concerns. Am I making sense?

Also, what if we don't need the first button i.e. we can't allow our user to review their recording and can't have a re-record option. So is there a way we can hide the first button and keep showing the 2nd button when `sendImmediately` is set to false? Or vice-versa.

**Comment by Bane-D** [comment]
Hi Shashank,

In general all of our videos are pushed up to our servers right away by default. This is the best way to make sure the videos reach your application and as such I would suggest to always use the default `.sendImmediately(true)`.

The `.sendImmediately(false)` was created for some specific cases where it is not that important if the videos get edited or sent later and offer the recording to be sent later on. This in general could lead to some videos being missed / unsent, removed by mistake and alike, however as mentioned in specific cases, this was needed.

If there is a need to add some specific buttons / offer some functionality and based on what you have mentioned so far I would suggest to [utilize events](https://github.com/Ziggeo/Android-Client-SDK#events--callbacks) instead.

Specifically "we can't allow our user to review their recording and can't have a re-record option" would be best satisfied by sending the videos to our servers as soon as they are recorded (default).

If you need to react to this in some way your code can know that the recording stopped with `void recordingStopped(@NonNull String path);` Plus if you want to utilize the `manuallySubmitted()` and that is the actual reason why you wanted to use `.sendImmediately(false)` then the following additional details might help.

Basically the manually submitted event would only tell you that this button was clicked at that point in time. We do not have any action nor requirement around this happening, so in all sense and for all purposes we can not tell you later if they did or did not press that button. This would be something you would react to and implement the needed setup on your own.

With that said, you can just as easily react to the recording being stopped at which point you could show a screen with a button that is more suited to your use case. Since you mentioned students and I think I remember you mentioned exam at some point, you could react to the recording stopped event with a screen that says:
"You are about to finish your exam. To do so click Finish button stop the clock"

As soon as someone clicks it you got their OK on the exam / manual approval or submission.

Hope this helps

**Comment by sagrawal31** [comment]
Hi, @Bane-D thank you for the long write-up and detailed explanation. 

I think I created the confusion here. Let me rephrase my question and concern (which I got some clarity after trying a few options and reading your comment).

Basically, I'm talking about these two buttons:

![image](https://user-images.githubusercontent.com/1804514/76510733-1aff1800-6478-11ea-9523-8a4f9948248f.png)

1. **Button 1**: "Stop & close button" which gives options to preview, re-record and submit the recording.
2. **Button 2**: "Send & close" check button which directly submits the recording.

I was able to hide these two buttons (or the entire control bar) by writing the [following code](https://github.com/wizpanda/cordova-plugin-ziggeo/blob/v1.0.0/src/android/com/ziggeo/CameraFullscreenRecorder.java#L127):

```
CameraRecorderStyle style = new CameraRecorderStyle.Builder()
    .hideControls(true)
    .build();

builder.style(style);
```

Earlier my question was focusing on the Button 1. Basically, I wanted to hide this particular button only because button 1 was showing the preview & re-record options and we don't want to give these options to our users.

Then after reading your comment & testing the combination of two options, I figured out that my configuration was setting `.sendImmediately(false)`. So once I set it to true (or removed it altogether because that is the default value), the behaviour of the Button 1 gets changed i.e. it also directly started to submit without giving options of preview or re-recording.

So, my original concern of raising this issue is kinda solved. So if you want, you can close this issue.

But as a UX perspective, I have a new concern. Basically, with the scenario of `.sendImmediately(true)`, both the buttons behave exactly the same i.e. they both stop the recording and starts uploading it to the servers.

So why to show the buttons 1 at all when `sendImmediately == true`?

I hope, this time I'm making sense 😄

**Comment by 3akat** [comment]
Hi Shashank ,
Yeah, it seems you're right in your last point :) Previously buttons behave in a little different way and they didn't do the same thing. If Bane agree I can remove Button1 (left) for `sendImmediately == true` case.

**Comment by sagrawal31** [comment]
Cool. Let @Bane-D take the decision.

**Comment by Bane-D** [comment]
Yes, that does make sense for the case of `sendImmediately(true)` is being used :) As my colleague mentioned, at a bit earlier time these buttons did different things and as upgrades were made, I guess one of those became just one too many :)

Thank you for reporting this Shashank :)

**Comment by sagrawal31** [comment]
Thanks @Bane-D 

@3akat please do this change as per your convenience. I'm renaming the issue title.

**Comment by 3akat** [comment]
Hi @sagrawal31, I've changed the behaviour in `0.82.7`, please take a look

**Comment by sagrawal31** [comment]
Thank you @3akat for considering this small UX issue.

This behaviour is now working as expected.
